packages/@aws-cdk/aws-lambda-python/test/integ.function.docker-volume.ts

```diff
@@ -0,0 +1,50 @@
+import { spawnSync } from 'child_process';
+import * as crypto from 'crypto';
+import * as path from 'path';
+import { Runtime } from '@aws-cdk/aws-lambda';
+import { App, Stack, StackProps } from '@aws-cdk/core';
+import { IntegTest, ExpectedResult } from '@aws-cdk/integ-tests';
+import { Construct } from 'constructs';
+import * as lambda from '../lib';
+
+/*
+ * Stack verification steps:
+ * * aws lambda invoke --function-name <deployed fn name> --invocation-type Event --payload '"OK"' response.json
+ */
+
+// We need to ensure the container name is unique and not re-used eg in parallel runs
+const containerName = crypto.randomBytes(12).toString('hex');
+
+// this is to simulate the situation that a volume can be mounted from another container
+spawnSync('docker', ['run', '-d', '--name', containerName, 'alpine', 'sleep 60']);
```

Comment:
I think we should set this up in a way that we actually use data from a volume of this other container. Otherwise I don't think the integ test doesn't really add much value.

Reply:
yeah, that would be ideal, but its slightly different to what would be happening in an actual case where the container executing CDK is the source container (which is the case for CICD) systems and my target use case.

I'll give it a try if it works with using the same paths as in the python bundling. But as they are hard coded and only allow for one source path I'm not sure if that is actually achievable in that level of detail.

What the current test at least provides is, that it ensures that during the bundling the volume of the temporary container can actually be mounted. It is not used then though.

Reply:
Ok, i do have an approach that _might_ work, but i run into race conditions to make the test actually work. It also needs modification of the bundling to include additional paths to be copied.
The spawn of the container itself is fast enough, that the bundle finds the correct container with the correct name. The actual execution and creation of the test data is not though, which means the copy fails as the directory does not yet exist.
Happy on any good pointers on how to achieve that, but i really want to avoid any "sleep" style things here to not add a flaky test

Reply:
Thanks for looking into this @webratz, I really appreciate it. Your investigations made me revised my initial view. I now think we actually do not want this test at all for all the reasons you've pointed out, plus:
- This code runs local only. Once it has reached CloudFormation it doesn't matter anymore
- `--volumes-from` is a docker option. As long as we are calling docker with it, we are good. Anything more is just testing docker.
- We don't have a test like this for `--volumes` either, for the same reasons.

Apologies for the additional work, but if you agree we can remove the test and I'm happy to exempt the PR from the requirement.

Reply:
i have it almost working :) will give it another 30 min to see if i can properly implement it with _some_ sleeps.
you can have a look then. otherwise of course also happy to remove it

Reply:
Impressive! Go ahead if you like and I will review then. But just to be transparent, I am kind of leaning towards this being not a useful test (and thus implicitly being unwanted in the sense that it adds to build time and potential for failure).

Reply:
pushed the tests, have a look at let me know if it should be kept or be gone.